libs/prelude/Language/Reflection/Elab.idr

```diff
@@ -510,6 +540,17 @@ namespace Tactics
   defineFunction : FunDefn Raw -> Elab ()
   defineFunction defun = Prim__DefineFunction defun
 
+  ||| Declare a datatype in the global context. This step only
```

Comment:
Where clauses ultimately turn into the global context with names that aren't accessible, and this can be used the same way.